rules/ast/ast.go

```diff
@@ -971,15 +1161,25 @@ func NewArithExpr(opType BinOpType, lhs Node, rhs Node) (Node, error) {
 		if lhs.Type() == ScalarType || rhs.Type() == ScalarType {
 			return nil, errors.New("AND and OR operators may only be used between vectors")
 		}
+		// logical operations must never be used with group modifiers.
+		if len(include) > 0 {
+			return nil, errors.New("AND and OR operators must not have a group modifier")
+		}
 	}
 
 	if lhs.Type() == VectorType || rhs.Type() == VectorType {
 		return &VectorArithExpr{
-			opType: opType,
-			lhs:    lhs,
-			rhs:    rhs,
+			opType:           opType,
+			lhs:              lhs,
+			rhs:              rhs,
+			matchCardinality: matchCard,
+			matchOn:          matchOn,
+			includeLabels:    include,
 		}, nil
 	}
+	if matchCard != MatchOneToOne || matchOn != nil || include != nil {
+		return nil, errors.New("binary scalar expressions cannot have vector matching options")
```

Comment:
I think this should also be disallowed if either side is a scalar. Currently the following don't result in an error:

    1 / on(y) group_left(x) a
    a / on(y) group_left(x) 1